githooks/post-merge

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+# === RUST-BITCOIN GITHOOK ===
+
+# Scan the githooks after merge. If the user is using any of the rust-bitcoin githooks and they have changed, let them know
+# to rerun the githooks setup.
+
+# We only care about this on master
+[[ "$(git branch --show-current)" = "master" ]] || exit 0
+
+# Get the local githooks directory, regardless of configuration.
+GIT_DIR=$(git rev-parse --git-common-dir)
+HOOKS_DIR=$(git config --get core.hooksPath || echo "$GIT_DIR/hooks")
+
+# Scan each of the hooks looking for the tag. If the tag is found, then the user is using a
+# rust-bitcoin githook. We should compare them to the repo's hooks to see if they have
+# changed in the latest git pull.
+for hook in "$HOOKS_DIR"/*
+do
+	if grep -q '=== RUST-BITCOIN GITHOOK ===' "$hook"; then
+		BN=$(basename "$hook")
+		if ! cmp --quiet "$hook" "githooks/$BN"; then
+			echo "========================================"
+			echo "Project githooks have changed."
+			echo "Please inspect the changes and re-reun"
+			echo "\`just githooks-install\` if they are"
+			echo "legitimate."
+			echo "========================================"
+			exit 1
```

Comment:
Why is it returning a non-zero exit code? I don't think it's an error, it's just warning.

Also I suspect we should output into stderr not stdout.

Reply:
My thinking with the exit code was that "this is abnormal", not that it was an error, per se.

Sort of related, I was also thinking that maybe there should be some kind of file that gets added, like `.scanned` which is created the first time a warning gets issued. If the script sees the file it doesn't issue the warning again. That way the user won't be pestered by the warning every time if they don't update their hooks. `just githooks-install` command could remove the file if it's there.

Reply:
I think it's completely normal and there's a good chance people will wait with it for a bit because they will not have time for it right away.

Reply:
Lots of commands exit with non-zero when it's not an error, such as grep, cmp, diff, etc. It's useful for automation/scripting when something can end in different states to have different exit codes.

>I think it's completely normal and there's a good chance people will wait with it for a bit because they will not have time for it right away.

Yeah, that's why I suggested the above solution where it only shows the above message the first time it notices that the scripts have changed.

Reply:
Yes but git hooks exiting with non-zero have special meaning. I don't know what the meaning of post merge is though.

I think reminding people periodically might be a good thing.

Reply:
I updated the PR with following changes:

1. No longer exits with a non-zero code after warning.
2. Outputs the warning to standard error.
3. Just trying this one on for size, let me know what you think: You can opt to ignore the warning manually if you want to keep the hooks and not see the warning on every pull.